packages/flash/src/index.ts

```diff
@@ -0,0 +1,93 @@
+import { AsyncLocalStorage } from 'node:async_hooks';
+import { Request, Response, NextFunction } from 'express';
+
+const als = new AsyncLocalStorage<FlashStorage>();
+
+export interface FlashMessage {
+  type: string;
+  message: string;
+}
+
+export function flashMiddleware() {
+  return (req: Request, _res: Response, next: NextFunction) => {
+    const flashStorage = makeFlashStorage(req);
+    als.run(flashStorage, () => next());
+  };
+}
+
+export function flash(): FlashMessage[];
+export function flash(type: string): string | null;
```

Comment:
Can we get some stronger typing on `type`?

Reply:
I was hesitant to constrain the keys here because there are desirable usages for flash data beyond just error/success/etc. messages. For instance, we could use this on the API token page to persist a newly-generated token for exactly one request.

This might of course be a case of YAGNI; I'm certainly not chomping at the bit to update our tokens page to use flash messages instead. 🤷

Reply:
If we ever do get around to using this for tokens (which I agree would be good) we could just add a `token` enum option at that point and handle it explicitly? If you do want to allow arbitrary strings here (I still don't see the use-case) then I'd suggest making that a `flashNonstandardType()` or something so that people will only use it if they really need it.

Reply:
OK, fair, YAGNI 😛 I'll make this change.